suite-native/atoms/src/Button/Button.tsx

```diff
@@ -1,118 +1,212 @@
+        onPressColor: 'green900',
+        disabledBackgroundColor: 'gray200',
+        textColor: 'gray0',
+        disabledTextColor: 'gray500',
+    },
+    tertiary: {
+        backgroundColor: 'gray200',
+        onPressColor: 'gray300',
+        disabledBackgroundColor: 'gray200',
+        textColor: 'gray800',
+        disabledTextColor: 'gray500',
+    },
+    danger: {
+        backgroundColor: 'red',
+        onPressColor: 'red700',
+        disabledBackgroundColor: 'gray200',
+        textColor: 'gray0',
+        disabledTextColor: 'gray500',
+    },
+} as const satisfies Record<ButtonColorSchemeName, ButtonColorScheme>;
 
-        const buttonColorSchemeStyles: Record<ButtonColorScheme, NativeStyleObject> = {
-            primary: {
-                backgroundColor: utils.colors.green,
-            },
-            gray: {
-                backgroundColor: utils.colors.gray300,
+const sizeToDimensionsMap = {
+    small: {
+        height: 40,
+        paddingVertical: 10,
+        paddingHorizontal: nativeSpacings.medium,
+    },
+    medium: {
+        height: 48,
+        paddingVertical: 12,
+        paddingHorizontal: 20,
+    },
+    large: {
+        height: 56,
+        paddingVertical: nativeSpacings.medium,
+        paddingHorizontal: nativeSpacings.large,
+    },
+} as const satisfies Record<ButtonSize, NativeStyleObject>;
+
+const textSizeToVariantMap = {
+    small: 'hint',
+    medium: 'body',
+    large: 'body',
+} as const satisfies Record<ButtonSize, TypographyStyle>;
+
+const iconStyle = prepareNativeStyle((utils, { position }: IconStyleProps) => ({
+    extend: [
+        {
+            condition: position === 'left',
+            style: {
+                marginRight: utils.spacings.small,
             },
-            red: {
-                backgroundColor: utils.transparentize(0.9, utils.colors.red),
+        },
+        {
+            condition: position === 'right',
+            style: {
+                marginLeft: utils.spacings.small,
             },
-        };
+        },
+    ],
+}));
 
+export const buttonStyle = prepareNativeStyle<ButtonStyleProps>(
+    (utils, { size, colorSchemeName, isDisabled }) => {
+        const sizeDimensions = sizeToDimensionsMap[size];
+        const schemeColors = buttonSchemeToColorsMap[colorSchemeName];
         return {
             flexDirection: 'row',
             justifyContent: 'center',
             alignItems: 'center',
             borderRadius: utils.borders.radii.round,
-            ...buttonSizeStyles[size],
-            ...buttonColorSchemeStyles[colorScheme],
-            extend: {
-                condition: isDisabled,
-                style: {
-                    backgroundColor: utils.colors.gray200,
+            ...sizeDimensions,
+            extend: [
+                {
+                    condition: isDisabled,
+                    style: {
+                        backgroundColor: utils.colors[schemeColors.disabledBackgroundColor],
```

Comment:
You forget to add `disabledTextColor`

Reply:
Forgot to add where? Animation? My assumption was that if the button is disabled it should not react to pressIn and pressOut events at all.

Reply:
Sorry, my bad I missed it set directly to Text in component